src/compiler/crystal/macros/methods.cr

```diff
@@ -422,7 +422,11 @@ module Crystal
       when "<="
         bool_bin_op(method, args) { |me, other| me <= other }
       when "<=>"
-        num_bin_op(method, args) { |me, other| me <=> other }
+        num_bin_op(method, args) do |me, other|
+          v = me <=> other
```

Comment:
Why not simply `(me <=> other) || NilLiteral.new`?

Reply:
Because `me <=> other` can return `false`?

Reply:
The expected return type is `Int32?`. Implementations could obviously return whatever they like, but that will likely break anyway.

Besides, this is compiler code and we control all implementations of `#<=>` that could be called here and none of them returns `false`.

Reply:
Done

Reply:
Note that `(me <=> other) || NilLiteral.new` didn't work, the `return` here returns from the method , not from the block. I had to use `(me <=> other) || (return NilLiteral.new)`.

In general I'd prefer if we didn't discuss these minor style things, they lead to commits that fail, revertion, discussions, etc., for very little gain.

Reply:
It was really just meant as a question because I didn't know whether the `return` was necessary or not. You didn't need to change that. Sorry.